edi_oca/models/edi_exchange_consumer_mixin.py

```diff
@@ -127,18 +132,18 @@ def get_view(self, view_id=None, view_type="form", **options):
                     "edi_oca.edi_exchange_consumer_mixin_buttons",
                     {"group": group},
                 )
-                node.addprevious(etree.fromstring(str_element))
-            View = self.env["ir.ui.view"]
-
-            # Override context for postprocessing
-            if view_id and res.get("base_model", self._name) != self._name:
-                View = View.with_context(base_model_name=res["base_model"])
-            new_arch, new_models = View.postprocess_and_fields(doc, self._name)
-            res["arch"] = new_arch
-            # We don't want to lose previous configuration, so, we only want to add
-            # the new fields
-            new_models.update(res["models"])
-            res["models"] = new_models
+                new_node = etree.fromstring(str_element)
+                new_arch, new_models = View.postprocess_and_fields(new_node, self._name)
+                for model in new_models:
+                    if model in all_models:
+                        continue
+                    if model not in res["models"]:
+                        all_models[model] = new_models[model]
+                    else:
```

Comment:
this `else` is superflous

Reply:
Do we really need this handling of the models?
I expect `models` to contain already the current model as you called super...
Just trying to understand how this works, sorry if I miss any piece of the puzzle :sweat_smile:

Reply:
Yes, but we need to ensure that no new models are added (for example on child records). For this reason we need to extend. The same logic is applied in base_tier_validation.
 I know that that should happen, but it is safer to keep it.